instrumentation/undertow-1.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/undertow/UndertowServerDispatchTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.undertow;
+
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.CAPTURE_HEADERS;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.ERROR;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.INDEXED_CHILD;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.QUERY_PARAM;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.REDIRECT;
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.SUCCESS;
+
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.undertow.Handlers;
+import io.undertow.Undertow;
+import io.undertow.util.Headers;
+import io.undertow.util.HttpString;
+import io.undertow.util.StatusCodes;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class UndertowServerDispatchTest extends AbstractHttpServerTest<Undertow> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpServerInstrumentationExtension.forAgent();
+
+  @Override
+  public Undertow setupServer() {
+    Undertow server =
+        Undertow.builder()
+            .addHttpListener(port, "localhost")
+            .setHandler(
+                Handlers.path()
+                    .addExactPath(
+                        SUCCESS.rawPath(),
+                        exchange ->
+                            exchange.dispatch(
+                                k ->
+                                    controller(
+                                        SUCCESS,
+                                        () -> {
+                                          k.getResponseSender().send(SUCCESS.getBody());
+                                          return null;
```

Comment:
I recently added an overload for `controller` that accepts `Runnable` instead of `Supplier`. If you rebase you can probably delete the `returns null`

Reply:
done